vllm/model_executor/tensorizer_loader.py

```diff
@@ -0,0 +1,233 @@
+    lazy_load: bool = False
+    plaid_mode_buffers: Optional[int] = None
+    verify_hash: bool = False
+    filter_func: Optional[Callable[[str], Union[bool, Any]]] = None
+    deserializer_encryption_key: Optional[str] = None
+    """
+  Args for TensorizerAgent class. These are used to configure the behavior of 
+  the TensorDeserializer when loading tensors from a serialized model.
+  
+  Args:
+      tensorizer_uri: Path to serialized model tensors. Can be a local file 
+          path or a S3 URI.
+      device: The device to load the tensors onto. If None, the tensors will be
+          loaded onto the default device.
+      dtype: The dtype to cast the tensors to. If None, the tensors will be 
+          loaded with their original dtype.
+      lazy_load: If True, tensors will be loaded and cached when keys are 
+          accessed. If False, all tensors will be loaded into memory up front.
+      plaid_mode_buffers: The number of buffers to use in plaid mode. This is 
+          only used if ``plaid_mode=True``. These buffers are used to 
+          pipeline the loading and processing of tensors.
+      verify_hash: If True, the hashes of each tensor will be verified against 
+          the hashes stored in the metadata. A `HashMismatchError` will be 
+          raised if any of the hashes do not match.
+      filter_func: A function that takes a tensor key and returns True if the 
+          tensor should be loaded and False if it should be skipped. If None,
+          all tensors will be loaded.
+      deserializer_encryption_key: A `DecryptionParams` object holding a 
+          password or key to use for decryption. ``None`` (the default) means 
+          no decryption.
+  """
+
+    def __post_init__(self):
+        self.file_obj = self.tensorizer_uri
+        self.s3_access_key_id = os.environ.get("S3_ACCESS_KEY_ID") or None
+        self.s3_secret_access_key = (os.environ.get("S3_SECRET_ACCESS_KEY")
+                                     or None)
+        self.s3_endpoint = os.environ.get("S3_ENDPOINT_URL") or None
+
+        self.stream_params = {
+            "s3_access_key_id": self.s3_access_key_id,
+            "s3_secret_access_key": self.s3_secret_access_key,
+            "s3_endpoint": self.s3_endpoint,
+            "force_http": True,
+        }
+
+        # Omitting self.dtype and self.device as this behaves weirdly
+        self.deserializer_params = {
+            "filter_func": self.filter_func,
+            "lazy_load": self.lazy_load,
+            "plaid_mode": bool(_is_vllm_model(file_uri=self.file_obj)),
+            "plaid_mode_buffers": self.plaid_mode_buffers,
+            "verify_hash": self.verify_hash,
+            "encryption": self.deserializer_encryption_key,
+            # "dtype":self.dtype,
+            # "device":self.device,
+        }
+
+    @staticmethod
+    def add_cli_args(
+        parser: argparse.ArgumentParser, ) -> argparse.ArgumentParser:
+        """Tensorizer CLI arguments"""
```

Comment:
Since these options are only valid for tensorizer, either their help text or layout in a CLI menu needs to communicate that requirement. This would be best represented by putting them in a tensorizer-specific [`argparse` argument group](https://docs.python.org/3/library/argparse.html#argument-groups).